tests/test_template.py

```diff
@@ -0,0 +1,200 @@
+from mira.metamodel import ControlledConversion, Concept, NaturalConversion
+
+
+def test_templates_equal():
+    infected = Concept(name='infected population',
+                       identifiers={'ido': '0000511'})
+    susceptible = Concept(name='susceptible population',
+                          identifiers={'ido': '0000514'})
+    c1 = ControlledConversion(
+        subject=Concept(name="susceptible"),
+        outcome=Concept(name="infected"),
+        controller=Concept(name="infected"),
+    )
+    c1_gnd = ControlledConversion(
+        subject=susceptible,
+        outcome=infected,
+        controller=infected,
+    )
+    c2 = ControlledConversion(
+        subject=Concept(name="susceptible"),
+        outcome=Concept(name="infected"),
+        controller=Concept(name="infected"),
+    )
+    c1_gnd_ctx = c1_gnd.with_context(location="Stockholm")
+    c2_ctx = c2.with_context(location="Stockholm")
+    assert not c1.is_equal_to(c2, with_context=False)
+    assert c1_gnd.is_equal_to(c1_gnd, with_context=False)
+    assert c1_gnd.is_equal_to(c1_gnd_ctx, with_context=False)
+    assert not c1_gnd.is_equal_to(c1_gnd_ctx, with_context=True)
+    assert not c1.is_equal_to(c2_ctx, with_context=True)
+    assert not c1.is_equal_to(c2_ctx, with_context=False)
+
+
+def test_concepts_equal():
+    c1 = Concept(name="infected population", identifiers={"ido": "0000511"})
+    c1_w_ctx = c1.with_context(location="Berlin")
+    c2 = Concept(name="infected population", identifiers={"ido": "0000511"})
+    c2_w_ctx = c2.with_context(location="Stockholm")
+
+    assert c1.is_equal_to(c2)
+    assert not c1_w_ctx.is_equal_to(c2_w_ctx, with_context=True)
+    assert c1_w_ctx.is_equal_to(c2_w_ctx, with_context=False)
+
+
+def test_template_type_inequality_is_equal():
+    infected = Concept(name="infected population", identifiers={"ido": "0000511"})
+    susceptible = Concept(name="susceptible population", identifiers={"ido": "0000514"})
+    immune = Concept(name="immune population", identifiers={"ido": "0000592"})
+    c1 = ControlledConversion(
+        subject=susceptible,
+        outcome=infected,
+        controller=infected,
+    )
+    n1 = NaturalConversion(subject=infected, outcome=immune)
+
+    try:
+        c1.is_equal_to(n1)
+    except Exception as exc:
+        assert isinstance(exc, TypeError), "Expected TypeError"
```

Comment:
Similar to my comment above, I don't think different template types should error when compared. Rather, the comparison should just return False.